gamla/functional_generic.py

```diff
@@ -208,3 +205,45 @@ def composition_over_binary_curried(*args):
         pair_with,
     )
 )
+
+
+_first_truthy_index = compose_left(
+    enumerate,
+    curried.filter(toolz.second),
```

Comment:
Is it ok to assume 0 is false here (as a value)?

Reply:
That's the usual signature of `filter`, anything truthy/falsy. The user can constraint this further by composing their filter function with a stricter condition.